src/BuildingRegistry/Building/Building_Grb.cs

```diff
@@ -110,5 +110,50 @@ public void Demolish(BuildingGrbData grbData)
                 buildingUnitsPosition));
             ApplyChange(new BuildingGeometryWasImportedFromGrb(BuildingPersistentLocalId, buildingGrbData));
         }
+
+        public void ChangeMeasurement(ExtendedWkbGeometry extendedWkbGeometry, BuildingGrbData grbData)
+        {
+            GuardRemovedBuilding();
+            GuardValidStatusses(BuildingStatus.Realized);
+
+            if (BuildingGeometry.Method != BuildingGeometryMethod.MeasuredByGrb)
+            {
+                throw new BuildingHasInvalidGeometryMethodException();
+            }
+
+            if (BuildingGeometry.Geometry == extendedWkbGeometry)
+            {
+                return;
+            }
+
+            var newBuildingGeometry = new BuildingGeometry(extendedWkbGeometry, BuildingGeometryMethod.MeasuredByGrb);
+            var plannedOrRealizedBuildingUnits = _buildingUnits.PlannedBuildingUnits()
+                .Concat(_buildingUnits.RealizedBuildingUnits())
+                .ToList();
+
+            var buildingUnitsOutsideOfBuildingMeasurement = plannedOrRealizedBuildingUnits
+                .Where(x =>
+                    x.BuildingUnitPosition.GeometryMethod == BuildingUnitPositionGeometryMethod.AppointedByAdministrator
+                    && !newBuildingGeometry.Contains(x.BuildingUnitPosition.Geometry))
+                .Select(x => x.BuildingUnitPersistentLocalId)
+                .ToList();
+
+            var buildingUnitsWithPositionDerivedFromBuilding = plannedOrRealizedBuildingUnits
+                .Where(x => x.BuildingUnitPosition.GeometryMethod == BuildingUnitPositionGeometryMethod.DerivedFromObject)
+                .Select(x => x.BuildingUnitPersistentLocalId)
+                .ToList();
+
+            var buildingUnitsPosition = buildingUnitsWithPositionDerivedFromBuilding.Any()
```

Comment:
I think you need an OR here: buildingUnitsWithPositionDerivedFromBuilding.Any() || **buildingUnitsOutsideOfBuildingMeasurement.Any()**